packages/vite/src/node/plugins/css.ts

```diff
@@ -1280,51 +1279,49 @@ async function compileCSS(
     return compileLightningCSS(id, code, environment, urlReplacer)
   }
 
+  const lang = CSS_LANGS_RE.exec(id)?.[1] as CssLang | undefined
+  const deps = new Set<string>()
+
+  // pre-processors: sass etc.
+  let preprocessorMap: ExistingRawSourceMap | undefined
+  if (isPreProcessor(lang)) {
+    const preprocessorResult = await compileCSSPreprocessors(
+      environment,
+      id,
+      lang,
+      code,
+      workerController,
+    )
+    code = preprocessorResult.code
+    preprocessorMap = preprocessorResult.map
+    preprocessorResult.deps?.forEach((dep) => deps.add(dep))
+  }
+
   const { modules: modulesOptions, devSourcemap } = config.css
   const isModule = modulesOptions !== false && cssModuleRE.test(id)
   // although at serve time it can work without processing, we do need to
   // crawl them in order to register watch dependencies.
   const needInlineImport = code.includes('@import')
   const hasUrl = cssUrlRE.test(code) || cssImageSetRE.test(code)
-  const lang = CSS_LANGS_RE.exec(id)?.[1] as CssLang | undefined
   const postcssConfig = await resolvePostcssConfig(
     environment.getTopLevelConfig(),
   )
 
-  // 1. plain css that needs no processing
+  // postcss processing is not needed
   if (
-    lang === 'css' &&
+    lang !== 'sss' &&
     !postcssConfig &&
     !isModule &&
     !needInlineImport &&
     !hasUrl
   ) {
-    return { code, map: null }
+    return { code, map: preprocessorMap ?? null, deps }
   }
```

Comment:
With this PR, If the preprocessor removed all `@import`, we might be able to return at this line. (previously postcss was run)